packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -268,12 +281,83 @@ class NetworkController with SearchControllerMixin<NetworkRequest> {
     // TODO(kenz): support intelligent search queries like t:http (type = http)
     // or m:GET (method = GET).
 
-    final currentRequests = _requests.value.requests;
+    final currentRequests = _filteredRequests.value;
     for (final request in currentRequests) {
       if (request.uri.toLowerCase().contains(caseInsensitiveSearch)) {
         matches.add(request);
       }
     }
     return matches;
   }
+
+  void filterData(NetworkFilter filter) {
+    if (filter == defaultFilter) {
+      _filteredRequests.value = List.from(_requests.value.requests);
+    }
+    _filteredRequests.value =
+        _requests.value.requests.where((NetworkRequest r) {
+      if (!filter.showHttp && r is HttpRequestData) {
+        return false;
+      }
+      if (!filter.showWebSocket && r is WebSocket) {
+        return false;
+      }
+      if (filter.method != null &&
+          r.method.toLowerCase() != filter.method.toLowerCase()) {
+        return false;
+      }
+      if (filter.uriSubstring != null &&
+          !r.uri.toLowerCase().contains(filter.uriSubstring.toLowerCase())) {
+        return false;
+      }
+      if (filter.status != null &&
+          r.status.toLowerCase() != filter.status.toLowerCase()) {
+        return false;
+      }
+      if (filter.type != null &&
+          r.type.toLowerCase() != filter.type.toLowerCase()) {
+        return false;
+      }
+      return true;
+    }).toList();
+    _activeFilter.value = filter;
+  }
+
+  void resetFilters() {
+    _activeFilter.value = defaultFilter;
+  }
+}
+
+class NetworkFilter {
+  NetworkFilter({
+    this.method,
+    this.uriSubstring,
+    this.status,
+    this.type,
+    this.showHttp = true,
+    this.showWebSocket = true,
+  });
+
+  static NetworkFilter from(NetworkFilter filter) {
+    return NetworkFilter(
+      method: filter.method,
+      uriSubstring: filter.uriSubstring,
+      status: filter.status,
+      type: filter.type,
+      showHttp: filter.showHttp,
+      showWebSocket: filter.showWebSocket,
+    );
+  }
+
+  String method;
```

Comment:
can these all be final?

Reply:
no because they need to be modified when adjusting the filter in the dialog. We could make a `copyWith` method. Leaving as is for now since I'll likely tweak all this code in a follow up to use a query.